pointmatcher/DataPointsFilters/Geometry.cpp

```diff
@@ -0,0 +1,170 @@
+	PointMatcher<T>::DataPointsFilter("GeometryDataPointsFilter",
+	        GeometryDataPointsFilter::availableParameters(), params),
+    keepUnstructureness(Parametrizable::get<int>("keepUnstructureness")),
+    keepStructureness(Parametrizable::get<T>("keepStructureness"))
+{
+}
+
+// Compute
+template<typename T>
+typename PointMatcher<T>::DataPoints
+GeometryDataPointsFilter<T>::filter(
+	const DataPoints& input)
+{
+	DataPoints output(input);
+	inPlaceFilter(output);
+	return output;
+}
+
+// In-place filter
+template<typename T>
+void GeometryDataPointsFilter<T>::inPlaceFilter(
+	DataPoints& cloud)
+{
+	typedef typename DataPoints::View View;
+	typedef typename DataPoints::Label Label;
+	typedef typename DataPoints::Labels Labels;
+
+	const int pointsCount(cloud.features.cols());
+	const int descDim(cloud.descriptors.rows());
+	const unsigned int labelDim(cloud.descriptorLabels.size());
+
+	// Check that the required eigenValue descriptor exists in the pointcloud
+    if (!cloud.descriptorExists("eigValues"))
+    {
+        throw InvalidField("GeometryDataPointsFilter: Error, no eigValues found in descriptors.");
+    }
+
+    // Validate descriptors and labels
+	int insertDim(0);
+	for(unsigned int i = 0; i < labelDim ; ++i)
+		insertDim += cloud.descriptorLabels[i].span;
+	if (insertDim != descDim)
+		throw InvalidField("SurfaceNormalDataPointsFilter: Error, descriptor labels do not match descriptor data");
+
+	// Reserve memory for new descriptors
+	const int dimSphericality(1);
+	const int dimUnstructureness(1);
+	const int dimStructureness(1);
+
+	boost::optional<View> sphericality;
+	boost::optional<View> unstructureness;
+	boost::optional<View> structureness;
+
+	Labels cloudLabels;
+	cloudLabels.push_back(Label("sphericality", dimSphericality));
+	if (keepUnstructureness)
+		cloudLabels.push_back(Label("unstructureness", dimUnstructureness));
+	if (keepStructureness)
+		cloudLabels.push_back(Label("structureness", dimStructureness));
+
+    // Reserve memory
+	cloud.allocateDescriptors(cloudLabels);
+
+	// Get the views
+    const View eigValues = cloud.getDescriptorViewByName("eigValues");
+    if (eigValues.rows() != 3)  // And check the dimensions
+    {
+        throw InvalidField("GeometryDataPointsFilter: Error, the number of eigValues is not 3.");
+    }
+
+	sphericality = cloud.getDescriptorViewByName("sphericality");
+	if (keepUnstructureness)
+		unstructureness = cloud.getDescriptorViewByName("unstructureness");
+	if (keepStructureness)
+		structureness = cloud.getDescriptorViewByName("structureness");
+
+    // Iterate through the point cloud and evaluate the geometry
+	for (int i = 0; i < pointsCount; ++i)
+	{
+	    // look at the three eigen values
+        Vector eig_vals_col = eigValues.col(i);
+        // might be already sorted but sort anyway
+        std::sort(eig_vals_col.data(),eig_vals_col.data()+eig_vals_col.size());
+
+        // finally, evaluate the geometry
+        T sphericality_val;
```

Comment:
These variables don't follow `camelCasing` for variables, as done in other lines of this file and libpointmatcher source.

Reply:
Fixed